File: lib_ebook_api/src/main/java/com/ebook/api/entity/Comment.java
package com.ebook.api.entity;

import android.os.Parcel;
import android.os.Parcelable;

public class Comment implements Parcelable {
    public static final Creator<Comment> CREATOR = new Creator<Comment>() {
        @Override
        public Comment createFromParcel(Parcel in) {
            return new Comment(in);
        }

        @Override
        public Comment[] newArray(int size) {
            return new Comment[size];
        }
    };
    private Long id;
    private User user;
    private String chapterUrl; //对应BookInfo noteUrl;
    private String chapterName;//当前章节名称
    private String bookName;
    private String comment;
    private String addtime;

    public Comment() {
    }

    protected Comment(Parcel in) {
        if (in.readByte() == 0) {
            id = null;
        } else {
            id = in.readLong();
        }
        chapterUrl = in.readString();
        chapterName = in.readString();
        bookName = in.readString();
        comment = in.readString();
        addtime = in.readString();
        user = in.readParcelable(Thread.currentThread().getContextClassLoader());
    }

    public String getAddtime() {
        return addtime;
    }

    public void setAddtime(String addtime) {
        this.addtime = addtime;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getChapterUrl() {
        return chapterUrl;
    }

    public void setChapterUrl(String chapterUrl) {
        this.chapterUrl = chapterUrl;
    }

    public String getChapterName() {
        return chapterName;
    }

    public void setChapterName(String chapterName) {
        this.chapterName = chapterName;
    }

    public String getBookName() {
        return bookName;
    }

    public void setBookName(String bookname) {
        this.bookName = bookname;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        if (id == null) {
            dest.writeByte((byte) 0);
        } else {
            dest.writeByte((byte) 1);
            dest.writeLong(id);
        }
        dest.writeString(chapterUrl);
        dest.writeString(chapterName);
        dest.writeString(bookName);
        dest.writeString(addtime);
        dest.writeParcelable(user, 0);
        dest.writeString(comment);
    }
}

File: lib_ebook_api/src/main/java/com/ebook/api/entity/Comment.kt
package com.ebook.api.entity

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * 评论 API 实体（对齐服务端评论契约：蛇形字段 + 章节冗余快照 + 分页包裹）。
 *
 * 双用途：创建评论请求体（[content] 必填，章节字段可选）+ 评论响应项。
 * 服务端载荷为蛇形命名，Kotlin 属性保持驼峰，边界翻译由 [SerialName] 完成：
 * - [commentKey] ↔ comment_key（评论聚合键，M2 新增）
 * - [chapterUrl]/[chapterName]/[bookName] ↔ chapter_url/chapter_name/book_name
 * - [content] ↔ content（内容字段，与旧契约的 comment 键不同）
 * - [addTime] ↔ add_time（Asia/Shanghai，yyyy-MM-dd HH:mm:ss）
 *
 * [user] 复用 [User] 实体解析服务端的评论用户视图（uid/username/nickname/avatar），
 * 服务端不返回 email，走默认值兜底。
 */
@Serializable
@Parcelize
data class Comment(
    var id: Long = 0L,
    @JvmField
    var user: User = User(),
    @SerialName("comment_key")
    var commentKey: String? = null, // 评论聚合键（M2）：ck1:hash 或 ck1:hash#章序号
    @SerialName("chapter_url")
    var chapterUrl: String? = null, //对应BookInfo noteUrl（已废弃，保留兼容）;
    @SerialName("chapter_name")
    var chapterName: String? = null, //当前章节名称
    @SerialName("book_name")
    var bookName: String? = null,
    @SerialName("content")
    var content: String? = null, //评论内容（创建评论必填）
    @SerialName("add_time")
    var addTime: String = ""
) : Parcelable
